src/sage/categories/simplicial_sets.py

```diff
@@ -574,6 +582,391 @@ def universal_cover(self):
+                    differentials[first-1] = matrix(base_ring, 0, 1)
+                    differentials[first] = matrix(base_ring, 1, rank,
+                                                [1] * rank)
+                else:
+                    differentials[first] = matrix(base_ring, 0, rank)
+
+                for d in dimensions:
+                    old_rank = rank
+                    faces = {_[1]: _[0] for _ in enumerate(current)}
+                    if d in simplices:
+                        current = sorted(simplices[d])
+                        rank = len(current)
+                        # old_rank: number of simplices in dimension d-1.
+                        # faces: list of simplices in dimension d-1.
+                        # rank: number of simplices in dimension d.
+                        # current: list of simplices in dimension d.
+                        if not faces:
+                            differentials[d] = matrix(base_ring, old_rank, rank)
+                        else:
+                            matrix_data = {}
+                            for col, sigma in enumerate(current):
+                                sign = 1
+                                twists = len(face_data[sigma]) * [1]
+                                twists[0] = twist(sigma)
+                                for (ch, tau) in zip(twists, face_data[sigma]):
+                                    if tau.is_nondegenerate():
+                                        row = faces[tau]
+                                        if (row, col) in matrix_data:
+                                            matrix_data[(row, col)] += sign*ch
+                                        else:
+                                            matrix_data[(row, col)] = sign*ch
+                                    sign *= -1
+
+                            differentials[d] = matrix(base_ring, old_rank,
+                                                    rank, matrix_data)
+
+                    else:
+                        rank = 0
+                        current = []
+                        differentials[d] = matrix(base_ring, old_rank, rank)
+
+                if cochain:
+                    new_diffs = {}
+                    for d in differentials:
+                        new_diffs[d-1] = differentials[d].transpose()
+                    return ChainComplex(new_diffs, degree_of_differential=1,
+                                        check=check)
+                return ChainComplex(differentials, degree_of_differential=-1,
+                                    check=check)
+
+            def twisted_homology(self, n, reduced=False):
+                r"""
+                The `n`-th twisted homology module of the simplicial set with respect to
+                the abelianization of the fundamental_group.
+
+                It is a module over a polynomial ring, including relations to make some
+                variables the multiplicative inverses of others.
```

Comment:
I think the option to express it in the Laurent polynomial rings should be done at some point.

Reply:
I agree, but in order for it to work properly, we would need to implement several functionalities in Laurent polynomials (involving matrices, modules, ideals....).

I think that should be done in a separate issue.